zebedee-cms/src/main/java/com/github/onsdigital/zebedee/session/store/JWTStore.java

```diff
@@ -70,29 +99,75 @@ public Session get(String id) throws IOException {
      */
     @Override
     public Session get(HttpServletRequest id) throws IOException {
-        return get();
+        info().log(GET_REQUEST_ID_NOOP);
+        return null;
     }
 
     /**
      * Get a {@link Session} session object from thread local.
-     *
+     *  
      * @param none.
      * @return session object from thread local.
      * @throws IOException for any problem getting a session from the request.
      */
     @Override
-    public Session get() throws IOException {
-        return null;
+    public ThreadLocal<UserDataPayload> get() throws IOException {
+        return store;
     }
 
     /**
      * Get a {@link Session} session object from thread local.
      *
-     * @param token/kid - the access token to be decoded, verified and stored and key id.
+     * @param token - the access token to be decoded.
      * @throws IOException for any problem verifying a token or storing a session in threadlocal.
      */
     @Override
-    public void set(String token, String kid) throws IOException {
+    public void set(String token) throws SessionsStoreException {
+        if (StringUtils.isEmpty(token)) {
+            throw new SessionsRequestException(ACCESS_TOKEN_REQUIRED_ERROR);
+        }
+
+        String[] chunks;
+        try {
+            chunks = token.split("\\.");
+        } catch (NullPointerException e) {
+            throw new SessionsDecodeException(TOKEN_NULL_ERROR, e);
+        }
+        // check token validity; throw error if []chunks doesn't contain 3 elements
+        if (chunks.length != JWT_CHUNK_SIZE) {
+            throw new SessionsDecodeException(TOKEN_NOT_VALID_ERROR);
+        }
+
+        String publicSigningKey = getPublicSigningKey(chunks[0]);
+
+        try {
+            store.set(
+                this.jwtHandler.verifyJWT(token, publicSigningKey)
+            );
+        } catch (JWTTokenExpiredException e) {
+            throw new SessionsTokenExpiredException(ACCESS_TOKEN_EXPIRED_ERROR);
+        } catch (JWTVerificationException e) {
+            throw new SessionsVerificationException(e.getMessage(), e);
+        } catch (JWTDecodeException e) {
+            throw new SessionsDecodeException(e.getMessage(), e);
+        }
+    }
+
+    private String getPublicSigningKey(String tokenHeader) {
+        String headerDecoded = new String(
+            Base64.getDecoder().decode(tokenHeader),
+            StandardCharsets.UTF_8
+        );
+
+        Map<String, String> decodedResult = new Gson().fromJson(
```

Comment:
Minor point this is likely to see high levels of throughput and constructing a new `Gson` each time is possibly expensive and time consuming.

Add a `Gson` as a class field and set it in the constructor - we can then use the same instance for each request.

Reply:
👍